src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx

```diff
@@ -255,6 +255,15 @@
   <data name="DoNotUseWeakCryptographicAlgorithmsMessage" xml:space="preserve">
     <value>{0} uses a weak cryptographic algorithm {1}</value>
   </data>
+  <data name="DoNotUseWeakKDFAlgorithm" xml:space="preserve">
+    <value>Do Not Use Weak Key Derivation Function Algorithm</value>
+  </data>
+  <data name="DoNotUseWeakKDFAlgorithmDescription" xml:space="preserve">
+    <value>Some implementations of the Rfc2898DeriveBytes class allow for a hashAlgorithm to be specified in the constructor parameter or overwritten in HashAlgorithm property. If this exists, then it should be used with SHA-256 or higher as the hashAlgorithm parameter.</value>
+  </data>
+  <data name="DoNotUseWeakKDFAlgorithmMessage" xml:space="preserve">
+    <value>{0} created without an approved hash algorithm. Use SHA256, SHA384 or SHA512 to create a strong key from password</value>
```

Comment:
{0} created with a weak hash algorithm.  Use SHA-256, SHA-384, or SHA-512 to create a strong key from a password.